knowledge graph/kg_agent.py

```
@@ -0,0 +1,326 @@
+            
+            result = await session.run(query, **properties)
+            record = await result.single()
+            return record['e'] if record else None
+    
+    async def create_relationship(self, from_type: str, from_props: Dict[str, Any], 
+                           rel_type: str, to_type: str, to_props: Dict[str, Any], 
+                           rel_props: Dict[str, Any] = None):
+        """Create a relationship between two entities"""
+        async with self.driver.session() as session:
+            # build match clauses for the entities
+            from_props_str = " AND ".join([f"a.{k} = ${k}" for k in from_props.keys()])
```

Ruff (0.12.2)

Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)